### DISCUSSION TITLE
Joining parent and child table

### DISCUSSION THREAD

**Discussion by bytesagar** [prisma/prisma#10345] [Q&A] [answered]
```prisma
model Patient {
  id Int @id @default(autoincrement())
  patient_id String
  name String
  caste String
  age Int
  gender String
  province Int
  district String
  palika String
  ward Int
  phone_number String
  vitals Vital?

}

model Vital {
  id Int @id @default(autoincrement())
  temperature Int
  pulse Int
  respiratory_rate Int
  blood_pressure Int
  spo_2 Int
  patient Patient @relation(fields: [patientId],references:[id])
  patientId Int

}
```

I want to have patient with the vital details in it.

**Comment by austincrim** [comment]
Hi @bytesagar 👋!

It sounds like you need to use our [`include` option](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#include).

```ts
const patientWithVitals = await prisma.patient.findUnique({
  where: { id },
  include: { vitals: true }
})
```

Let me know if that works for you!

**Reply by bytesagar** [reply]
Hi @austincrim 
I am sorry but my query was little different. How can i post vital form data. Should i use prisma.patient.update or prisma.vitals.create
I tried this
```ts
const addVitals = async (req,res) => {
    console.log(req.params)
    const {id} = req.params
    const {temp,pulse,rr,bp,spo_2} = req.body
    const patient = await prisma.patient.update({
        data : {
            vitals: {
                temperature: temp,
                pulse: pulse,
                respiratory_rate: rr,
                blood_pressure: bp,
                spo_2: spo_2,
                connect: {
                    patientId: id
                }
            }
        }
    })

    res.json(patient)
}
```
 but its throwing me error

**Comment by austincrim** [comment, marked as answer]
@bytesagar 

If you want to create a `vital` through the `patient` model, you will need to use one of our [nested write](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#nested-writes) APIs.

Here's an example using a nested `create`.
```ts
const patient = await prisma.patient.update({
    data: {
        vitals: {
            create: {
              temperature: temp,
              pulse: pulse,
              respiratory_rate: rr,
              blood_pressure: bp,
              spo_2: spo_2,
            }
        }
    },
    where: { id: 123 }
})
```

You also have to pass a `where` property when using the `update` function.

Let me know if that doesn't work!

**Reply by bytesagar** [reply]
@austincrim 

I tried this it worked too

```
const patient = await prisma.vital.create({
    data: {
      temperature: temp,
      pulse: pulse,
      respiratory_rate: rr,
      blood_pressure: bp,
      spo_2: spo_2,
      patient: {
        connect: {
          patient_id: id,
        },
      },
    },
  });

  res.json(patient);
};

```
It creates vitals with user id on it and updates user vitals.
Is it good approach or should I shift to your method.

**Reply by bytesagar** [reply]
@austincrim 

So it worked but, there is something weird going on that when i post the vitals for the first time patient vital gets updated, but at the same time when i want to again update the patient with new vitals then it throws ` Null constraint violation on the fields: (`patientId`)` this error. 
How can I solve this ?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
